evennia/objects/manager.py

```diff
@@ -575,8 +575,8 @@ def copy_object(
             return None
 
         # copy over all attributes from old to new.
-        for attr in original_object.attributes.all():
-            new_object.attributes.add(attr.key, attr.value)
+        attrs = ((a.key, a.value, a.category, a.lock_storage) for a in original_object.attributes.all())
+        new_object.attributes.batch_add(*attrs)
```

Comment:
This is a good optimization :+1: I don't think the `attributes.batch_add` existed when the Attribute manager was originally created :)